src/main/java/taskitems/TaskManager.java

```diff
@@ -0,0 +1,236 @@
+        greet.printDecoration();
+        saveData();
+    }
+
+    private void addToTasks(String taskName, boolean isBackend) throws IllegalInputException {
+        if (taskName.equals("")) {
+            throw new IllegalInputException();
+        }
+        tasks.add(new Todo(taskName));
+        taskCount++;
+    }
+
+    private void addToTasks(String type, String taskName,String date, boolean isBackend){
+        if (type.equals("E")) {
+            tasks.add(new Event(taskName, date));
+        } else {
+            tasks.add(new Deadline(taskName, date));
+        }
+        taskCount++;
+    }
+
+    private void markTask(int number, boolean isBackend) throws IllegalInputException {
+        if (number > taskCount || number < 1) {
+            throw new IllegalInputException();
+        } else {
+            tasks.get(number - 1).setMarked(true);
+        }
+    }
+
+    public void loadData() throws FileNotFoundException {
+        greet.printDecoration();
+        System.out.println("***Please wait while I fetch past data...***");
+        File data = new File(PATH);
+        Scanner reader = new Scanner(data);
+        int count = 1;
+        while (reader.hasNext()) {
+            String line = reader.nextLine();
+            String[] dataLine = line.split(",");
+            boolean hasFailed = false;
+            switch (dataLine[0]) {
+            case "T":
+                try {
+                    addToTasks(dataLine[1],true);
+                } catch (IllegalInputException e) {
+                    System.out.println("Line " + count + " is corrupted.");
+                    hasFailed = true;
+                }
+                break;
+            case "D":
+                try {
+                    addToTasks("D",dataLine[1],dataLine[3],true);
+                } catch (ArrayIndexOutOfBoundsException outOfBoundsException) {
+                    System.out.println("Line " + count + " is corrupted.");
+                    hasFailed = true;
+                }
+                break;
+            case "E":
+                try {
+                    addToTasks("E",dataLine[1],dataLine[3],true);
+                } catch (ArrayIndexOutOfBoundsException outOfBoundsException) {
+                    System.out.println("Line " + count + " is corrupted.");
+                    hasFailed = true;
+                }
+                break;
+            default:
+                System.out.println("Line " + count + " is corrupted,");
+                hasFailed = true;
+                break;
+            }
+            if (!hasFailed && dataLine[2].equals("1")) {
+                try {
+                    markTask(taskCount,true);
+                } catch (IllegalInputException e) {
+                    System.out.println("Corrupted data has been loaded, please check task number: " + taskCount);
+                }
+            }
+            count++;
+        }
+        if (taskCount == 0) {
```

Comment:
You could make the [happy path more prominent](https://nus-cs2113-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#make-the-happy-path-prominent) in `loadData()` - show is done with valid tasks using [SLAP](https://nus-cs2113-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#slap-hard) and also use a guard clause for the path for invalid or no task data found.